queries.sql

```diff
@@ -37,4 +37,14 @@ SELECT COUNT(*) FROM ANIMALS WHERE ESCAPE_ATTEMPTS = 0;
 SELECT AVG(WEIGHT_KG) AS AVERAGE_WEIGHT FROM ANIMALS;
 SELECT NEUTERED AS NEUTERED_ANIMALS, MAX(ESCAPE_ATTEMPTS) AS ANIMALS_ESCAPE_THE_MOST FROM ANIMALS GROUP BY NEUTERED;
 SELECT SPECIES, MAX(WEIGHT_KG), MIN(WEIGHT_KG) FROM ANIMALS GROUP BY SPECIES;  
-SELECT SPECIES, AVG(ESCAPE_ATTEMPTS) FROM ANIMALS WHERE DATE_OF_BIRTH BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY SPECIES;
\ No newline at end of file
+SELECT SPECIES, AVG(ESCAPE_ATTEMPTS) FROM ANIMALS WHERE DATE_OF_BIRTH BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY SPECIES;
+
+SELECT ANIMALS.NAME AS ANIMAL, VETS.NAME, DATE_OF_VISIT FROM ANIMALS JOIN VISITS ON ANIMALS.ID = VISITS.ANIMALS_ID JOIN VETS ON VISITS.VETS_ID = VETS.ID WHERE VETS.NAME = 'William Tatcher' ORDER BY DATE_OF_VISIT DESC LIMIT 1;
+SELECT COUNT(VISITS.ANIMALS_ID), VETS.NAME FROM VISITS JOIN VETS ON VISITS.VETS_ID = VETS.ID WHERE VETS.NAME = 'Stephanie Mendez' GROUP BY VETS.NAME; 
+SELECT VETS.NAME, SPECIES.NAME FROM VETS LEFT JOIN SPECIALIZATIONS ON VETS.ID = SPECIALIZATIONS.VETS_ID LEFT JOIN SPECIES ON SPECIALIZATIONS.SPECIES_ID = SPECIES.ID;
+SELECT ANIMALS.NAME, VETS.NAME, VISITS.DATE_OF_VISIT FROM ANIMALS JOIN VISITS ON ANIMALS.ID = VISITS.ANIMALS_ID JOIN VETS ON  VISITS.VETS_ID = VETS.ID WHERE VETS.NAME = 'Stephanie Mendez' AND DATE_OF_VISIT BETWEEN '2020-03-01' AND '2020-08-30';
+SELECT ANIMALS.NAME, COUNT(ANIMALS.NAME) FROM ANIMALS JOIN VISITS ON ANIMALS.ID = VISITS.ANIMALS_ID GROUP BY ANIMALS.NAME ORDER BY COUNT DESC LIMIT 1;
+SELECT ANIMALS.NAME, VETS.NAME, VISITS.DATE_OF_VISIT FROM ANIMALS JOIN VISITS ON ANIMALS.ID = VISITS.ANIMALS_ID JOIN VETS ON VISITS.VETS_ID = VETS.ID WHERE VETS.NAME = 'Maisy Smith' ORDER BY DATE_OF_VISIT ASC LIMIT 1;
+SELECT ANIMALS.NAME, VETS.NAME, VISITS.DATE_OF_VISIT FROM ANIMALS JOIN VISITS ON ANIMALS.ID = VISITS.ANIMALS_ID JOIN VETS ON VISITS.VETS_ID = VETS.ID ORDER BY DATE_OF_VISIT DESC LIMIT 1;
+SELECT COUNT(*) FROM VETS JOIN VISITS ON VETS.ID = VISITS.VETS_ID JOIN ANIMALS ON VISITS.ANIMALS_ID = ANIMALS.ID LEFT JOIN SPECIALIZATIONS ON ANIMALS.SPECIES_ID = SPECIALIZATIONS.SPECIES_ID AND VETS.ID = SPECIALIZATIONS.VETS_ID WHERE SPECIALIZATIONS.SPECIES_ID IS NULL;
+SELECT SPECIES.NAME AS SPECIALTY FROM ANIMALS JOIN VISITS ON ANIMALS.ID = VISITS.ANIMALS_ID JOIN VETS ON VISITS.VETS_ID = VETS.ID JOIN SPECIES ON ANIMALS.SPECIES_ID = SPECIES.ID WHERE VETS.NAME = 'Maisy Smith' GROUP BY SPECIES.NAME ORDER BY COUNT(*) DESC LIMIT 1;
```

Comment:
Kindly note that all the `JOIN(s)` implemented here should be `LEFT JOIN` instead of just normal `JOIN` to combine rows that matches in both tables 😄